### DISCUSSION TITLE
How to checkout at the path for every step?

### DISCUSSION THREAD

**Discussion by nilsreichardt** [community/community#25697] [Actions]
<p>My current workflow looks something like:</p>
<pre><code class="lang-auto">jobs:
  test:
    if: github.event.pull_request.draft == false
    runs-on: ubuntu-latest
    env:
      DEV_KEY: ${{ secrets.FIREBASE_ADMIN_SDK_DEV_KEY }}
    steps:
    - uses: actions/checkout@v2
    - name: Install dependencies
      run: |
        cd infrastructure/google_cloud_platform/firebase/cloud_functions/typescript_cloud_functions/functions
        npm install
    - name: Lint
      run: |
        cd infrastructure/google_cloud_platform/firebase/cloud_functions/typescript_cloud_functions/functions
        npm run lint
    - name: Check for compiling errors
      run: |
        cd infrastructure/google_cloud_platform/firebase/cloud_functions/typescript_cloud_functions/functions
        npm run build
    - name: Run Unit Tests
      run: |
        cd infrastructure/google_cloud_platform/firebase/cloud_functions/typescript_cloud_functions/functions
        npm test
    - name: Run E2E Tests
      run: |
        cd infrastructure/google_cloud_platform/firebase/cloud_functions/typescript_cloud_functions/functions
        echo $DEV_KEY &gt; dev-key.json
        npm run test_e2e
</code></pre>
<p>Is there a way to start every time at <code>infrastructure/google_cloud_platform/firebase/cloud_functions/typescript_cloud_functions/functions</code>? There is a <code>path</code> parameter for <code>checkout</code>, but it seems so that this is not the thing I’m searching for</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>I think <a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#defaultsrun"><code>defaults.run.working-directory</code></a> should do what you need.</p>

**Comment by nilsreichardt** [comment]
<p>Yes, this works :+1:  Thank you! :pray:</p>
